### PR TITLE
EVG-2238 add configuration and data model for display tasks

### PR DESCRIPTION
**Pull request by john-m-liu**
The display task model adds 2 new fields for whether it's a display task and also what execution tasks it contains. For the other task fields, my approach was to add it to the display task document if it doesn't change between its execution tasks (project, build...) , and compute it from the execution tasks if it can be different (status, priority...)

### CODE REVIEW

File: model/project.go [hunk 1]
```diff
@@ -387,17 +393,16 @@ func (tt TaskIdTable) GetIdsForAllTasks(currentVariant, taskName string) []strin
 }
 
 // TaskIdTable builds a TaskIdTable for the given version and project
-func NewTaskIdTable(p *Project, v *version.Version) TaskIdTable {
+func NewTaskIdTable(p *Project, v *version.Version) (TaskIdTable, TaskIdTable) {
```
Comment (tychoish) [inline model/project.go]: I'd rather have this return a struct with two fields one for display and one for execution that would make it less likely that someone was going to get the order backwards.

File: model/project.go [hunk 2]
```diff
@@ -406,15 +411,28 @@ func NewTaskIdTable(p *Project, v *version.Version) TaskIdTable {
...
 		}
 	}
-	return table
+	return execTable, displayTable
 }
 
 // NewPatchTaskIdTable constructs a new TaskIdTable (map of [variant, task display name]->[task id])
-func NewPatchTaskIdTable(proj *Project, v *version.Version, patchConfig TVPairSet) TaskIdTable {
-	table := TaskIdTable{}
+func NewPatchTaskIdTable(proj *Project, v *version.Version, patchConfig TVPairSet) (TaskIdTable, TaskIdTable) {
```
Comment (tychoish) [inline model/project.go]: same as above.

File: model/project.go (hunk 1 above)
Comment (john-m-liu) [reply model/project.go]: done

File: model/project.go (hunk 2 above)
Comment (john-m-liu) [reply model/project.go]: done

File: model/lifecycle.go [hunk 3]
```diff
@@ -486,7 +489,7 @@ func CreateBuildFromVersion(project *Project, v *version.Version, tt TaskIdTable
 // The slice of tasks will be in the same order as the project's specified tasks
 // appear in the specified build variant.
 func createTasksForBuild(project *Project, buildVariant *BuildVariant,
```
Comment (tychoish) [inline model/lifecycle.go]: can this also take the new type

Comment (john-m-liu) [reply model/lifecycle.go]: done

File: model/task/db.go [hunk 4]
```diff
@@ -381,6 +382,14 @@ func ByDispatchedWithIdsVersionAndStatus(taskIds []string, versionId string, sta
 	})
 }
 
+func ByExecutionTask(taskId string) db.Q {
+	return db.Query(bson.M{
+		ExecutionTasksKey: bson.M{
+			"$in": []string{taskId},
```
Comment (tychoish) [inline model/task/db.go]: shouldn't this be an equality query? there's only one value?

Comment (john-m-liu) [reply model/task/db.go]: That field is an array of values, so equality won't work. Let me know if there's a cleaner way to query for a string array containing a specific string.

**Comment by john-m-liu** [comment]
I still need to implement tags in display tasks, but could we push this and I'll implement that with the next ticket?

**Pull request merged**